### ISSUE TITLE
Q: Compute complete Betweenness Centrality

### ISSUE THREAD

**Issue by delanyinspiron6400** [hornet-gt/hornet#54]
TL:DR: What is the difference between the three variants of BC and how can one compute the complete BC for a graph (like in [Hybrid_BC](https://github.com/Adam27X/hybrid_BC))?

I have a few questions regarding computing the BC of a graph.
It seems there are three variants in the repository:
- `BCCentrality`
- `ExactBC`
- `ApproximateBC`

What is the difference between these three variants, a few of them are out-commented in the `BCTest.cu`, hence I'm not completely sure if they are working as of now?

Furthermore, as far as I can see, these algorithms seem to require a root node to be passed, does this mean that I'd have to sequentially call the algorithms for all nodes in a graph to actually compute the BC?
I guess internally, some form of Brandes is done, hence does this mean that by passing a root node, only a partial BC result is computed, looking at all paths from this given root node?
Could I then simply do something like this:
```cpp
BCCentrality bc(hornet_graph);
for(auto i = 0; i < graph.nV(), ++i)
{
  bc.setRoot(i);
  bc.run();
}
```
or is there a canonical approach to compute the BC for a graph I'm missing?

Thanks very much! :smile:

**Comment by ogreen** [comment]
Can you repost your question here:
https://github.com/rapidsai/cuhornet
The above is a modified repo with newer code than the one here.

**Comment by delanyinspiron6400** [comment]
Sure, I'll do so :)
Could you maybe note that in the repository somewhere at the top (as is done with the hornetsnest repository) such that one can more easily find the current version, thanks.
